src/config.cpp

```diff
@@ -687,10 +697,53 @@ void parse_program_options(int argc, const char** argv)
             std::exit(EXIT_FAILURE);
         }
 
+        try
+        {
+            config.metric_leader =
+                perf::EventProvider::get_event_by_name(config.metric_leader).name;
+        }
+        catch (const perf::EventProvider::InvalidEvent& e)
+        {
+            Log::fatal() << "The requested --metric-leader is invalid";
+            std::exit(EXIT_FAILURE);
+        }
+
         config.metric_use_frequency = false;
         config.metric_count = arguments.as<std::uint64_t>("metric-count");
     }
 
+    const auto& leader_cpus =
+        perf::EventProvider::get_event_by_name(config.metric_leader).get_supported_cpus();
+
+    if (arguments.given("standard-metrics"))
+    {
+        for (const auto& mem_event : platform::get_mem_events())
+        {
+            config.perf_group_events.emplace_back(mem_event.name);
+        }
+        config.perf_group_events.emplace_back("instructions");
```

Comment:
We have these fancy `EventDescription` class, why do we only store strings in `config.perf_group_events`. I'd bet somewhere else in the code we use a bunch of `get_event_by_name` calls to translate all event names in this thing again.